packages/grpc-js/src/http_proxy.ts

```diff
@@ -202,9 +205,15 @@ export function getProxiedConnection(
             ' through proxy ' +
             proxyAddressString
         );
-        resolve({
-          socket,
-          realTarget,
+        var cts = tls.connect({
+            ...connectionOptions,
+            host: options.host,
```

Comment:
`options.host` here is the IP address of the proxy, and when `socket` is set, `host` is only used to validate the server certificates. Why would you want to validate the server certificates against the proxy's IP address?

Reply:
Good question. Also works without setting `host` - have pushed update that removes.

Reply:
Actually, I think you want `host: getDefaultAuthority(realTarget)` here, so that you don't have to pass the `grpc.ssl_target_name_override` option when constructing the client. In fact, TLS certificate validation is the whole reason we pass the default authority to `http2.connect` in the first place.

Reply:
updated